benchmark/conversions/conversions.cpp

```diff
@@ -0,0 +1,270 @@
+        << "Usage: " << (*argv)[0] << " [options]\n"
+        << "  The standard input should contain a list of test cases as a JSON "
+        << "array of objects:\n"
+        << "  [\n"
+        << "    { \"filename\": \"my_file.mtx\"},\n"
+        << "    { \"filename\": \"my_file2.mtx\"}\n"
+        << "  ]\n\n"
+        << "  The results are written on standard output, in the same format,\n"
+        << "  but with test cases extended to include an additional member \n"
+        << "  object for each conversion run in the benchmark.\n"
+        << "  If run with a --backup flag, an intermediate result is written \n"
+        << "  to a file in the same format. The backup file can be used as \n"
+        << "  input \n to this test suite, and the benchmarking will \n"
+        << "  continue from the point where the backup file was created.";
+
+    gflags::SetUsageMessage(doc.str());
+    std::ostringstream ver;
+    ver << gko::version_info::get();
+    gflags::SetVersionString(ver.str());
+    gflags::ParseCommandLineFlags(argc, argv, true);
+}
+
+
+// matrix format creating mapping
+template <typename MatrixType>
+std::unique_ptr<gko::LinOp> read_matrix(
+    std::shared_ptr<const gko::Executor> exec, const gko::matrix_data<> &data)
+{
+    auto mat = MatrixType::create(std::move(exec));
+    mat->read(data);
+    return mat;
+}
+
+
+#define READ_MATRIX(MATRIX_TYPE, ...)                                   \
+    [](std::shared_ptr<const gko::Executor> exec,                       \
+       const gko::matrix_data<> &data) -> std::unique_ptr<gko::LinOp> { \
+        auto mat = MATRIX_TYPE::create(std::move(exec), __VA_ARGS__);   \
+        mat->read(data);                                                \
+        return mat;                                                     \
+    }
+
+
+const std::map<std::string, std::function<std::unique_ptr<gko::LinOp>(
+                                std::shared_ptr<const gko::Executor>,
+                                const gko::matrix_data<> &)>>
+    matrix_factory{
+        {"csr", READ_MATRIX(csr, std::make_shared<csr::automatical>())},
+        {"coo", read_matrix<gko::matrix::Coo<>>},
+        {"ell", read_matrix<gko::matrix::Ell<>>},
+        {"hybrid", read_matrix<hybrid>},
+        {"sellp", read_matrix<gko::matrix::Sellp<>>}};
+
+
+template <typename RandomEngine>
+void convert_matrix(const gko::LinOp *matrix_from, const gko::LinOp *matrix_to,
+                    const char *conversion_name,
+                    std::shared_ptr<gko::Executor> exec,
+                    const gko::matrix_data<etype> &data,
+                    const unsigned int warm_iter, const unsigned int run_iter,
+                    rapidjson::Value &test_case,
+                    rapidjson::MemoryPoolAllocator<> &allocator,
+                    RandomEngine &engine) try {
+    auto &conversion_case = test_case["conversions"];
+    if (!FLAGS_overwrite && conversion_case.HasMember(conversion_name)) {
```

Comment:
`FLAGS_overwrite` is the only `FLAGS` parameter that you use from the global variables (at least I think it is one), the others (`FLAGS_warmup` & `FLAGS_repetitions` in lines 255+), you pass as a parameters.

I think you can delete this `if` since you do basically the same thing in lines 247+ with a `continue`, so in that case, this function should not be called. If you want to be on the safe side and keep the `if`, I would prefer a function parameter for `FLAGS_overwrite`.